### ISSUE TITLE
Change of azurerm_kubernetes_cluster.default_node_pool.vm_size destroys the cluster

### ISSUE THREAD

**Issue by jorgecarleitao** [terraform-providers/terraform-provider-azurerm#7231]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.23
azurerm = "~> 1.36"
```

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "main" {
  name                = var.name
  location            = data.azurerm_resource_group.main.location
  resource_group_name = data.azurerm_resource_group.main.name
  dns_prefix          = var.name

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_B2s"
  }

  service_principal {
    client_id     = var.client_id
    client_secret = var.client_secret
  }

  addon_profile {
    http_application_routing {
      enabled = false
    }
    kube_dashboard {
      enabled = true
    }
  }
}
```

### Expected Behavior

The default pool should have been destroyed, and a new one should have been created.

### Actual Behavior

The whole cluster was destroyed, and a new one was created.

### Steps to Reproduce

1. `terraform apply -auto-approve`
2. change `Standard_B2s` to `Standard_D11_v2`
3. `terraform apply -auto-approve`
4. Watch the whole cluster crash and burn after step 3 and a new one reborn.

### Important Factoids

There is nothing [in the documentation](https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html#default_node_pool) mentioning that changing the default_node_pool causes the resource to be destroyed.

**Comment by tombuildsstuff** [comment]
hi @jorgecarleitao 

Thanks for opening this issue :)

Taking a look through this appears to be a duplicate of #7093 - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one; would you mind subscribing to #7093 for updates?

Thanks!

**Comment by jorgecarleitao** [comment]
Woops, sorry @tombuildsstuff ! Yes, let's work it out on that one.
